### DISCUSSION TITLE
How to express "not exists property" in c# document query

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#13423] [Q&A] [answered]
What would be the correct syntax for the "not exist property" query

from Collection where Property = null or not exists(Property)

when using C# document query. The following example is not enough (the result is empty list):

var result = session.Query<Collection>().Where(x => x.Property == null).ToList()

**Comment by garayx** [comment, marked as answer]
Hi 

You can use the `WhereExists` method which is available in `DocumentQuery`, see:
https://ravendb.net/docs/article-page/5.3/Csharp/client-api/session/querying/how-to-filter-by-field#examples

**Comment by ben-jacobs** [comment]
Not sure if this is still the case and the c# API may do this correctly, but in previous versions you couldn't just use _not exists_ on it's own when writing RQL etc. - you need to do 

` .... where true and not exists x`

i.e. the not exists check can't be on its own.

**Reply by ayende** [reply]
The C# client handles that for you
